pulsar-client-api/src/main/java/org/apache/pulsar/client/api/DeadLetterPolicy.java

```diff
@@ -60,4 +60,9 @@ public class DeadLetterPolicy {
      * to be created.
      */
     private String initialSubscriptionName;
+
+    /**
+     * Name of the dead letter producer name.
+     */
+    private String producerName;
```

Comment:
This is a change for the public API. We need a PIP to get approved to add this field.